File: frontend/src/services/lectureSummaryService.js
import apiClient from './apiClient';
import { API_PATHS } from '../constants/apiPaths';

/**
 * Service module for post-lecture summary API calls.
 */

/** Teacher: submits a summary for a specific completed lecture. */
export const submitSummary = (lectureId, payload) => {
  return apiClient.post(`${API_PATHS.LECTURE_SUMMARIES}/${lectureId}`, payload);
};

/** Teacher: fetches the summary for a specific lecture. */
export const getMySummaryForLecture = (lectureId) => {
  return apiClient.get(`${API_PATHS.LECTURE_SUMMARIES}/${lectureId}/me`);
};

/** Teacher: page of own summaries, newest first. */
export const getMySummaries = (page = 0, size = 10) => {
  return apiClient.get(API_PATHS.LECTURE_SUMMARIES_ME, { params: { page, size } });
};

/** Admin: all summaries for lectures on a given day (null = today). */
export const getSummariesByDate = (date = null, page = 0, size = 20) => {
  const params = { page, size };
  if (date) params.date = date;
  return apiClient.get(API_PATHS.LECTURE_SUMMARIES, { params });
};

/** Admin: all summaries whose lecture date falls within [from, to] — backs monthly export. */
export const getSummariesByRange = (from, to, page = 0, size = 100) => {
  return apiClient.get(API_PATHS.LECTURE_SUMMARIES, { params: { from, to, page, size } });
};

export const lectureSummaryService = {
  submitSummary,
  getMySummaryForLecture,
  getMySummaries,
  getSummariesByDate,
  getSummariesByRange,
};

File: Backend/src/main/java/com/institute/workforce_tracking/service/LectureSummaryService.java
package com.institute.workforce_tracking.service;

import java.time.LocalDate;

import com.institute.workforce_tracking.dto.request.SubmitLectureSummaryRequest;
import com.institute.workforce_tracking.dto.response.LectureSummaryResponse;
import com.institute.workforce_tracking.dto.response.PagedResponse;

/**
 * Business logic for post-lecture summaries, required within 24 hours of
 * lecture completion. Missing summaries trigger lecture cancellation.
 */
public interface LectureSummaryService {

    /**
     * Submits the caller's summary for a specific completed lecture.
     * Throws BadRequest if the lecture is not COMPLETED or a summary was
     * already submitted.
     */
    LectureSummaryResponse submitSummary(String teacherEmail, Long lectureId,
                                         SubmitLectureSummaryRequest request);

    /** The caller's summary for a specific lecture (404 if none). */
    LectureSummaryResponse getMySummaryForLecture(String teacherEmail, Long lectureId);

    /** A page of the caller's summaries, newest first. */
    PagedResponse<LectureSummaryResponse> getMySummaries(String teacherEmail, int page, int size);

    /** Manager view: all summaries for lectures on one day. */
    PagedResponse<LectureSummaryResponse> getSummariesByDate(LocalDate date, int page, int size);

    /** Manager view: all summaries for lectures within [from, to] — backs the monthly export. */
    PagedResponse<LectureSummaryResponse> getSummariesByDateRange(
            LocalDate from, LocalDate to, int page, int size);

    /**
     * Scheduler-invoked sweep: cancels completed lectures whose actual end
     * was ≥24h ago without a submitted summary (teachers only).
     * Returns count of lectures cancelled.
     */
    int cancelLecturesWithMissingSummaries();
}
